ArduCopter/GCS_Mavlink.cpp

```diff
@@ -786,6 +786,9 @@ MAV_RESULT GCS_MAVLINK_Copter::handle_command_long_packet(const mavlink_command_
                 (int8_t)packet.param3,
                 is_positive(packet.param4));
             return MAV_RESULT_ACCEPTED;
+        } else if (packet.param1 <= -1.0f) {
+            copter.flightmode->auto_yaw.set_mode(AUTO_YAW_LOOK_AHEAD);
```

Comment:
_"maybe it needs to be different depending upon the flight mode"_  
I used **autopilot_yaw_mode Mode::AutoYaw::default_mode** to implement the logic. I had to change the bool parameter to mode parameter to keep the caller part **copter.flightmode->auto_yaw.set_mode_to_default(copter.flightmode->mode_number());** clearer.

_"My only other concern is just that -1 could be interpreted as 1deg left of north but I think the mavlink message spec says it must be in the range of 0 to 360 so that's ok actually."_ 
GCS_Mavlink strictly test **if ((packet.param1 >= 0.0f)**.
if this part is approved we may need to update it in the [documentation](https://mavlink.io/en/messages/common.html).